plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -51,32 +58,72 @@ export const EntityPicker = (props: EntityPickerProps) => {
     (allowedKinds && { kind: allowedKinds });
 
   const defaultKind = uiSchema['ui:options']?.defaultKind;
-  const defaultNamespace = uiSchema['ui:options']?.defaultNamespace;
+  const defaultNamespace =
+    uiSchema['ui:options']?.defaultNamespace || undefined;
 
   const catalogApi = useApi(catalogApiRef);
 
-  const { value: entities, loading } = useAsync(() =>
-    catalogApi.getEntities(
+  const { value: entities, loading } = useAsync(async () => {
+    const { items } = await catalogApi.getEntities(
       catalogFilter ? { filter: catalogFilter } : undefined,
-    ),
-  );
+    );
+    return items;
+  });
+  const allowArbitraryValues =
+    uiSchema['ui:options']?.allowArbitraryValues ?? true;
 
-  const entityRefs = entities?.items.map(e =>
-    humanizeEntityRef(e, { defaultKind, defaultNamespace }),
+  const getLabel = useCallback(
+    (ref: string) => {
+      try {
+        return humanizeEntityRef(
+          parseEntityRef(ref, { defaultKind, defaultNamespace }),
+          {
+            defaultKind,
+            defaultNamespace,
+          },
+        );
+      } catch (err) {
+        return ref;
+      }
+    },
+    [defaultKind, defaultNamespace],
   );
 
   const onSelect = useCallback(
-    (_: any, value: string | null) => {
-      onChange(value ?? undefined);
+    (_: any, ref: string | Entity | null, reason: AutocompleteChangeReason) => {
+      // ref can either be a string from free solo entry or
+      if (typeof ref !== 'string') {
+        onChange(ref ? stringifyEntityRef(ref as Entity) : '');
```

Comment:
@benjdlambert this would actually be fine for backwards compatibility right? given that `humanizeEntityRef` is able to produce a full entity ref anyway

Reply:
Yep think that this is fine.

Reply:
Alright, gonna ship this as is so we can try it out in the release 👍